components/script/dom/gamepad.rs

```diff
@@ -143,10 +193,76 @@ impl Gamepad {
         self.index.set(index);
     }
 
+    pub fn update_timestamp(&self, timestamp: f64) {
+        self.timestamp.set(timestamp);
+    }
+
     pub fn notify_event(&self, event_type: GamepadEventType) {
         let event = GamepadEvent::new_with_type(&self.global(), event_type, &self);
         event
             .upcast::<Event>()
             .fire(self.global().as_window().upcast::<EventTarget>());
```

Comment:
I believe we also need to update according to https://w3c.github.io/gamepad/#extensions-to-the-windoweventhandlers-interface-mixin

See `dom/bindings/webidl/EventHandler.webidl`